stl/inc/sstream

```diff
@@ -71,27 +110,53 @@ public:
     }
 
     enum { // constants for bits in stream state
-        _Allocated = 1, // set if character array storage has been allocated, eback() points to allocated storage
-        _Constant  = 2, // set if character array nonmutable
-        _Noread    = 4, // set if character array cannot be read
-        _Append    = 8, // set if all writes are appends
-        _Atend     = 16 // set if initial writes are appends
+        _Allocated   = 1, // set if character array storage has been allocated, eback() points to allocated storage
+        _Constant    = 2, // set if character array nonmutable
+        _Noread      = 4, // set if character array cannot be read
+        _Append      = 8, // set if all writes are appends
+        _Atend       = 16, // set if initial writes are appends
+        _From_rvalue = 32 // set if character array is released from an rvalue of basic_string
     };
 
     using int_type = typename _Traits::int_type;
     using pos_type = typename _Traits::pos_type;
     using off_type = typename _Traits::off_type;
 
-    _NODISCARD _Mystr str() const {
-        _Mystr _Result(_Al);
-        if (!(_Mystate & _Constant) && _Mysb::pptr()) { // writable, make string from write buffer
+    struct _Buffer_view {
+        _Elem* _Ptr;
+        _Mysize_type _Size;
+        _Mysize_type _Res;
+    };
+
+    _NODISCARD _Buffer_view _Get_buffer_view() const noexcept {
+        _Buffer_view _Result{};
+        if ((!(_Mystate & _Constant) || (_Mystate & _From_rvalue)) && _Mysb::pptr()) {

```

Comment:
Agree. I don't think I broke ABI, because I didn't change any virtual methods or members.

Reply:
I remain slightly concerned and mix-and-match issues here - specifically when mixing code using this feature (i.e. new toolset in C++20 mode) and code compiled with an old toolset that wasn't aware of the new bit in the flags.

However, since I can't point to an actual scenario of badness, and it shouldn't be possible for this to break automatically during an upgrade (someone would deliberately have to be compiling `/std:c++latest` and then use this specific feature, *and* be mixing older TUs), I think the risk is acceptable.